src/internal/connector/graph/service.go

```diff
@@ -264,6 +259,14 @@ func kiotaMiddlewares(
 		khttp.NewParametersNameDecodingHandler(),
 		khttp.NewUserAgentHandler(),
 		&LoggingMiddleware{},
+	}...)
+
+	// Optionally add concurrency limiter middleware if it has been initialized.
+	if concurrencyLim != nil {
+		mw = append(mw, concurrencyLim)
+	}
+
+	mw = append(mw, []khttp.Middleware{
 		&RateLimiterMiddleware{},
 		&MetricsMiddleware{},
 	}...)
```

Comment:
`slice...` isn't needed here, you can just add the individual values. 
```suggestion
	mw = append(
	        mw,
		&RateLimiterMiddleware{},
		&MetricsMiddleware{})
```